processing/src/main/java/io/druid/query/aggregation/DoubleMaxAggregatorFactory.java

```diff
@@ -131,15 +156,17 @@ public String getName()
   @Override
   public List<String> requiredFields()
   {
-    return Arrays.asList(fieldName);
+    return fieldName != null ? Arrays.asList(fieldName) : Parser.findRequiredBindings(fieldExpression);
   }
 
   @Override
   public byte[] getCacheKey()
   {
-    byte[] fieldNameBytes = StringUtils.toUtf8(fieldName);
+    byte[] fieldNameBytes = StringUtils.toUtf8WithNullToEmpty(fieldName);
+    byte[] fieldExpressionBytes = StringUtils.toUtf8WithNullToEmpty(fieldExpression);
 
-    return ByteBuffer.allocate(1 + fieldNameBytes.length).put(CACHE_TYPE_ID).put(fieldNameBytes).array();
+    return ByteBuffer.allocate(1 + fieldNameBytes.length + fieldExpressionBytes.length)
```

Comment:
Should place 0xff between these guys, in the event that a `fieldName` is used that matches a `fieldExpression` but should have different results.